applications/asset_tracker_v2/src/cloud/cloud_codec/cloud_codec.h

```diff
@@ -255,29 +259,94 @@ enum cloud_codec_event_type {
 };
 
 struct cloud_codec_evt {
+	/** Cloud codec event type */
 	enum cloud_codec_event_type type;
+	/** New config data if type is CLOUD_CODEC_EVT_CONFIG_UPDATE */
 	struct cloud_data_cfg config_update;
 };
 
+/** @brief Event handler prototype.
+ *
+ * @param[in] evt event type
+ */
 typedef void (*cloud_codec_evt_handler_t)(const struct cloud_codec_evt *evt);
 
+/**
+ * @brief Initialize cloud codec.
+ * @details Is called once in the data module to initialize the cloud codec used.
+ * @param[in] cfg initial config data
+ * @param[in] event_handler handler for events coming from the codec
+ *
+ * @note currently only used for config updates in lwm2m
+ *
+ * @return 0 on success or error code according to errno.h.
+ */
 int cloud_codec_init(struct cloud_data_cfg *cfg, cloud_codec_evt_handler_t event_handler);
 
+/**
+ * @brief Encode cloud codec neighbor cells data.
+ * @details Is called as part of the data_encode() function to encode LTE neighbor cells info.
```

Comment:
Gives specific information about the caller. I don't think this is necessary. The docs should be kept as generic as possible.